Built QuestionPaper backend schema

arhamkac opened the pull request:
Resolves #19 

## Description

I built question paper schema and started logs schema

## Live Demo (if any)
[
<img width="828" height="501" alt="Screenshot from 2025-09-06 23-53-15" src="https://github.com/user-attachments/assets/05c6483f-6139-4df5-a5f8-4e27443f18a9" />
]

## Checkout
- [x] I have read all the contributor guidelines for the repo.

MrImmortal09 commented:
@0PrashantYadav0 @arhamkac  cloudinary provides 5GB of storage ig, do u have any other better options?

0PrashantYadav0 commented:
@MrImmortal09 I think this will work fine for now. After we grow we will change provider or have new plans.

0PrashantYadav0 commented on helpers/cloudinary.js:
```diff
@@ -0,0 +1,40 @@
+import {v2 as cloudinary} from "cloudinary";
+import fs from "fs";
+
+cloudinary.config({
+    cloud_name:process.env.CLOUDINARY_CLOUD_NAME,
+    api_key:process.env.CLOUDINARY_API_KEY,
+    api_secret:process.env.CLOUDINARY_API_SECRET
+})
```
@arhamkac Make these changes

0PrashantYadav0 commented on helpers/cloudinary.js:
```diff
@@ -0,0 +1,40 @@
+
+const deleteOnCloudinary=async(publicId)=>{
+    try {
+        if(!publicId){return;}
```
Use proper linting everywhere ( refer other files )

0PrashantYadav0 commented on model/paper.ts:
```diff
@@ -0,0 +1,57 @@
+import mongoose,{Schema,Document,Model} from "mongoose";
+
+export interface IPaper extends Document{
+    title:string,
+    content:string,
+    subject:string,
+    year:Date,
+    term:string,
+    document_url?:string,
+    uploaded_by?:string
+}
```
Apply these changes here

0PrashantYadav0 commented on model/paper.ts:
```diff
@@ -0,0 +1,57 @@
+            type:String,
+            required:[true,"Title of question paper is required"],
+            trim:true,
+            maxlength:[100,"Title must be withing 100 words"]
+        },
```
this one also

0PrashantYadav0 commented on model/paper.ts:
```diff
@@ -0,0 +1,57 @@
+            required:[true,"Please enter subject name for given paper"],
+            trim:true
+        },
+        year:{
+            type:Date,
+            required:[true,"Year of question paper is necessary"],
+            max:new Date().getFullYear()
+        },
```
if you are using data make sure to have proper type everywhere.

arhamkac commented:
Ok I will make these changes @0PrashantYadav0  and keep theses things in mind

MrImmortal09 commented:
@arhamkac whenever you feel that you are ready with the PR, do mention me or prashant, and also its not neccessary to resolve all the comments made by the bot, the bot will even find errors in a perfect code, so just give a look at it and apply only if necessary.

arhamkac commented:
Ok @MrImmortal09 I will once again check all the functionalities and if they all work fine will tell you

arhamkac commented:
I have done all the necessary changes. Now you can review and tell if any necessary changes are required @MrImmortal09  and @0PrashantYadav0 . Also for ease I have added a video for you to verify how everything is working.

https://github.com/user-attachments/assets/9b1ef947-6965-468b-b240-eb84c38ac229

And for logs I have also added them. Here are some screenshots of them in database-
<img width="367" height="348" alt="Logs-1" src="https://github.com/user-attachments/assets/b2a120ca-7832-4548-b3e4-9a0b6ba4ab37" />

<img width="517" height="307" alt="Logs-2" src="https://github.com/user-attachments/assets/12179a46-4e1d-472b-8fb9-253fe00c9ecd" />

arhamkac commented:
> @arhamkac are u still working on this??

Ya just the build issue is there, rest is solved. Sorry for the delay. I will see it today

MrImmortal09 reviewed (approved):
The build is failing as environment variables for cloudinary are not added to vercel , I will add them directly in production, rest fix the changes requested by @0PrashantYadav0 and we are good to go with this PR.
